src/log.cpp

```diff
@@ -310,16 +310,30 @@ void Logger::logToOutputs(LogLevel lev, const std::string &combined,
 //// *LogOutput methods
 ////
 
-void FileLogOutput::open(const std::string &filename)
+void FileLogOutput::setFile(const std::string &filename, u64 file_size_max)
 {
-	m_stream.open(filename.c_str(), std::ios::app | std::ios::ate);
+	actionstream << "Log messages are saved to " << filename << std::endl;
+
+	std::ifstream ifile(filename, std::ios::binary | std::ios::ate);
+	bool is_too_large = ifile.tellg() > static_cast<std::streamoff>(file_size_max);
+	ifile.close();
+
+	if (is_too_large) {
+		std::string filename_secondary = filename + ".1";
+		actionstream << "The log file grew too big; it is moved to " <<
+			filename_secondary << std::endl;
+		std::remove(filename_secondary.c_str());
+		std::rename(filename.c_str(), filename_secondary.c_str());
```

Comment:
The existing code usually doesn't prefix C standard functions with `std::`

Reply:
I've changed it. Unfortunately there are no native C++ functions for file renaming and removal as far as I knot.

Reply:
There are, but they are C++17 and newer.
https://en.cppreference.com/w/cpp/filesystem/remove